pkg/resources/management.cattle.io/v3/authconfig/validator.go

```diff
@@ -0,0 +1,272 @@
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return gvr
+}
+
+// Operations returns list of operations handled by the validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Update, admissionregistrationv1.Create}
+}
+
+// ValidatingWebhook returns the ValidatingWebhook.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) []admissionregistrationv1.ValidatingWebhook {
+	return []admissionregistrationv1.ValidatingWebhook{
+		*admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.ClusterScope, v.Operations()),
+	}
+}
+
+// Admitters returns the admitter objects.
+func (v *Validator) Admitters() []admission.Admitter {
+	return []admission.Admitter{&v.admitter}
+}
+
+type admitter struct {
+}
+
+// Admit handles the webhook admission requests.
+func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	listTrace := trace.New("authconfigValidator Admit", trace.Field{Key: "user", Value: request.UserInfo.Username})
+	defer listTrace.LogIfLong(admission.SlowTraceDuration)
+
+	oldAuthConfig, newAuthConfig, err := objectsv3.AuthConfigOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AuthConfig from request: %w", err)
+	}
+
+	switch request.Operation {
+	case admissionv1.Create:
+		return a.admitCreate(request, newAuthConfig)
+	case admissionv1.Update:
+		return a.admitUpdate(request, oldAuthConfig, newAuthConfig)
+	default:
+		return admission.ResponseAllowed(), nil
+	}
+}
+
+func (a *admitter) admitCreate(request *admission.Request, newAuthConfig *v3.AuthConfig) (*admissionv1.AdmissionResponse, error) {
+	return a.admitCommonCreateUpdate(request, nil, newAuthConfig)
+}
+
+func (a *admitter) admitUpdate(request *admission.Request, oldAuthConfig, newAuthConfig *v3.AuthConfig) (*admissionv1.AdmissionResponse, error) {
+	return a.admitCommonCreateUpdate(request, oldAuthConfig, newAuthConfig)
+}
+
+func (a *admitter) admitCommonCreateUpdate(request *admission.Request, _, newAuthConfig *v3.AuthConfig) (*admissionv1.AdmissionResponse, error) {
+	var err error
+
+	switch newAuthConfig.Type {
+	case "openLdapConfig", "freeIpaConfig":
+		err = validateLDAPConfig(request)
+	case "activeDirectoryConfig":
+		err = validateActiveDirectoryConfig(request)
+	default:
+	}
+
+	if err != nil {
+		return admission.ResponseBadRequest(err.Error()), nil
+	}
+
+	return admission.ResponseAllowed(), nil
+}
+
+func validateLDAPConfig(request *admission.Request) error {
+	var (
+		err    error
+		config v3.LdapConfig
+	)
+
+	if err = json.Unmarshal(request.Object.Raw, &config); err != nil {
+		return fmt.Errorf("failed to get %T from request: %w", config, err)
+	}
+
+	if len(config.Servers) < 1 {
+		err = errors.Join(err, field.Forbidden(field.NewPath("servers"), "at least one server is required"))
+	}
```

Comment:
FYI This check we have in testAndApply on the Rancher side, that's why it was included.